### PR TITLE
Event search

### PR DESCRIPTION
**Pull request by guainumbi**
Adds an event search bar to the event index page, so users can search for and find specific events faster.

### CODE REVIEW

**Comment by backspace** [comment]
Hey, @guainumbi, us Travisites are looking into some mysterious failures to trigger builds, it looks like it’s affecting your PR here. We don’t know what’s going on but I thought I’d let you know it’s Not You 😆

Hopefully we can figure out something soon, I’m sorry for the interruption!

**Comment by guainumbi** [comment]
Haha alright, thanks for letting me know @backspace. 😄 👍

**Comment by backspace** [comment]
You’re welcome. We haven’t figured out the cause of why this started happening BUT you may find that builds at least run now! 🤞🏻

**Comment by guainumbi** [comment]
hello @backspace I am not sure but I think something is still not running smoothly... My local tests all run without errors or failures. 🤔Do you have any idea what could be the problem?

**Comment by backspace** [comment]
hmm I can’t explain why it would work locally unless you have a different database setup but it appears to me based on the errors that the problem is that the database isn’t being cleared between tests. In Ruby tests, people often use something like [Database Cleaner](https://github.com/DatabaseCleaner/database_cleaner) to delete everything after a test has run so there’s no bleedthrough to the next test.

[This error](https://travis-ci.org/rubymonsters/diversity_ticketing/builds/425810808#L912-L915) for instance:

```
ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "tags_pkey"
DETAIL:  Key (id)=(1) already exists.
: INSERT INTO "tags" ("id", "name", "created_at", "updated_at", "category_id") VALUES ($1, $2, $3, $4, $5) RETURNING "id"
    test/services/event_search_service_test.rb:8:in `block (3 levels) in <class:EventSearchServiceTest>'
```

In [your tests](https://github.com/rubymonsters/diversity_ticketing/pull/444/files#diff-51fb2e7301608944ecba7ba0c2dfbd96), you’re constructing tags with explicit ids like 1 and 2 and then looking for those ids in the assertions. Since an earlier test makes a tag with id 1, a later test can’t unless everything has been reset. One way to avoid this would be to change your test to look something like this:

```
       tag = Tag.create!(name: 'Ruby', category: Category.create!(name: 'Programming language'))
       assert_equal EventSearchService.new(query: '', filter: { tag_ids: [tag.id] }).results, []
```

Leaving out the id lets the database choose one itself, and then you can use the generated id in the assertion. I think that’s the typical approach to testing for ids so you don’t run into these primary key uniqueness constraints.

Let me know if this makes sense?

File: app/assets/javascripts/dropdown.js [hunk 1]
```diff
@@ -14,4 +14,10 @@ $(document).ready(function(){
       }
     }
   });
+
```
Comment (mauditecandela) [inline app/assets/javascripts/dropdown.js]: I would create a separate dropdown.js for the filter dropdown. Like in language-picker.js

**Pull request merged**